Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.IntegrationTests/ApiGatewayResponseExtensionsSpecialCases.cs

```diff
@@ -0,0 +1,115 @@
+            var lambdaClient = new AmazonLambdaClient(RegionEndpoint.USWest2);
+            var iamClient = new AmazonIdentityManagementServiceClient(RegionEndpoint.USWest2);
+
+            _helper = new ApiGatewayTestHelper(null, apiGatewayV2Client, lambdaClient, iamClient);
+            _httpClient = new HttpClient();
+        }
+
+        public async Task InitializeAsync()
+        {
+            // Create IAM Role for Lambda
+            _roleArn = await _helper.CreateIamRoleAsync();
+
+            // Create Lambda function
+            var lambdaCode = @"
+            exports.handler = async (event, context, callback) => {
+                console.log(event);
+                callback(null, event.body);
+            };";
+            _lambdaArn = await _helper.CreateLambdaFunctionAsync(_roleArn, lambdaCode);
+
+            // Create HTTP API (v2)
+            (_httpApiV2Id, _httpApiV2Url) = await _helper.CreateHttpApi(_lambdaArn, "2.0");
+
+            // Wait for the API Gateway to propagate
+            await Task.Delay(10000); // Wait for 10 seconds
+
+            // Grant API Gateway permission to invoke Lambda
+            await _helper.GrantApiGatewayPermissionToLambda(_lambdaArn);
+        }
+
+        [Fact]
+        public async Task V2_TestCanInferJsonType()
+        {
+
+            var testResponse = new APIGatewayHttpApiV2ProxyResponse
+            {
+                Body = "Hello from lambda" // a regular string is considered json in api gateway
+            };
+
+            var httpTestResponse = testResponse.ToHttpResponse();
+            var actualResponse = await _httpClient.PostAsync(_httpApiV2Url, new StringContent("Hello from lambda"));
+
+            await _helper.AssertResponsesEqual(actualResponse, httpTestResponse);
+            Assert.Equal(200, (int)actualResponse.StatusCode);
+            Assert.Equal("application/json", actualResponse.Content.Headers.ContentType.ToString());
+            var content = await actualResponse.Content.ReadAsStringAsync();
+            Assert.Equal("Hello from lambda", content);
+        }
+
+        [Fact]
+        public async Task V2_TestCanInferJsonType2()
+        {
+
+            var testResponse = new APIGatewayHttpApiV2ProxyResponse
+            {
+                Body = "{\"key\" : \"value\"}"
+            };
+
+            var httpTestResponse = testResponse.ToHttpResponse();
+            var actualResponse = await _httpClient.PostAsync(_httpApiV2Url, new StringContent("{\"key\" : \"value\"}"));
+
+            await _helper.AssertResponsesEqual(actualResponse, httpTestResponse);
+            Assert.Equal(200, (int)actualResponse.StatusCode);
+            Assert.Equal("application/json", actualResponse.Content.Headers.ContentType.ToString());
+            var content = await actualResponse.Content.ReadAsStringAsync();
+            Assert.Equal("{\"key\" : \"value\"}", content);
+        }
+
+        [Fact]
+        public async Task V2_HandlesNonJsonResponse()
```

Comment:
this test case actually fails as of now - in reality it still returns "{key}". ive kept it here to comment on how i dont really understand api gateways behavior in this case